### PR TITLE
Prometheus for scheduler

### PR DESCRIPTION
**Pull request by valiksinev**

### CODE REVIEW

File: v2-advanced/clickhouse-scheduler/Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,4 @@ log = "0.4.17"
 anyhow = "1.0.70"
 tryhard = "0.5.0"
 fast_log = { version = "1.5.54" }
+prometheus-client = "0.19.0"
```
Comment (Deniskore) [inline v2-advanced/clickhouse-scheduler/Cargo.toml:21]: Please update to 0.20.0

File: v2-advanced/clickhouse-scheduler/src/executor.rs [hunk 2]
```diff
@@ -113,17 +132,19 @@ async fn execute_with_retry(
...
+pub async fn start_tasks(config: Arc<Config>, shutdown: Receiver<()>, task_metrics: Vec<TaskMetric>  ) {
     let mut set = JoinSet::new();
 
     let clients = Arc::new(ClickHouse::from_config(&config));
     let tasks = config.tasks.clone();
 
-    for task in tasks {
+    for (task, metric) in tasks.iter().zip(task_metrics) {
+        let task = task.clone();
+        // let metric = metric.clone();
```
Comment (Deniskore) [inline v2-advanced/clickhouse-scheduler/src/executor.rs:143]: Please remove comment.

**Review by Deniskore — CHANGES_REQUESTED** [review]
Please also run cargo clippy

```
warning: using `clone` on type `f64` which implements the `Copy` trait
  --> v2-advanced/clickhouse-scheduler/src/executor.rs:71:41
   |
71 |                     metric.min_time.set(measurement_queue.iter().min_by( |a, b| a.total_cmp(b)).unwrap_or(&0_f64).clone());
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try dereferencing it: `*measurement_queue.iter().min_by( |a, b| a.total_cmp(b)).unwrap_or(&0_f64)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy
   = note: `#[warn(clippy::clone_on_copy)]` on by default

warning: using `clone` on type `f64` which implements the `Copy` trait
  --> v2-advanced/clickhouse-scheduler/src/executor.rs:72:41
   |
72 |                     metric.max_time.set(measurement_queue.iter().max_by( |a, b| a.total_cmp(b)).unwrap_or(&0_f64).clone());
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try dereferencing it: `*measurement_queue.iter().max_by( |a, b| a.total_cmp(b)).unwrap_or(&0_f64)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: unneeded `return` statement
   --> v2-advanced/clickhouse-scheduler/src/executor.rs:115:5
    |
115 |     return None
    |     ^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
    = note: `#[warn(clippy::needless_return)]` on by default
    = help: remove `return`

warning: redundant clone
  --> v2-advanced/clickhouse-scheduler/src/prometheus.rs:45:72
   |
45 |             registry.register(&errors_name, &errors_name, metric.errors.clone());
note: this value is dropped without further use
  --> v2-advanced/clickhouse-scheduler/src/prometheus.rs:45:59
   |
45 |             registry.register(&errors_name, &errors_name, metric.errors.clone());
   |                                                           ^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
   = note: `#[warn(clippy::redundant_clone)]` on by default

warning: `clickhouse-scheduler` (bin "clickhouse-scheduler") generated 4 warnings
```

File: v2-advanced/clickhouse-scheduler/src/executor.rs (hunk 2 above)
Comment (valiksinev) [reply v2-advanced/clickhouse-scheduler/src/executor.rs:143]: comment removed

File: v2-advanced/clickhouse-scheduler/Cargo.toml (hunk 1 above)
Comment (valiksinev) [reply v2-advanced/clickhouse-scheduler/Cargo.toml:21]: version updated

**Review by Deniskore — APPROVED** [review]

**Pull request merged**